### PR TITLE
Add alerting API.

### PR DESCRIPTION
**Pull request by Ma-Dan**

### CODE REVIEW

**Comment by rayzhou2017** [comment]
Defer to 2.1.0

File: pkg/apis/alerting/v1alpha2/register.go [hunk 1]
```diff
@@ -0,0 +1,1380 @@
...
+		Param(ws.QueryParameter("sort_key", "Sort key. One of t1.alert_id, t1.alert_name, t1.disabled, t1.running_status, t1.alert_status, t1.create_time, t1.update_time, t1.policy_id, t1.rs_filter_id, t1.executor_id, t2.policy_id, t2.policy_name, t2.policy_description, t2.policy_config, t2.creator, t2.available_start_time, t2.available_end_time, t2.create_time, t2.update_time, t2.rs_type_id, t3.rs_filter_id, t3.rs_filter_name, t3.rs_filter_param, t3.status, t3.create_time, t3.update_time, t3.rs_type_id, t4.rs_type_id, t4.rs_type_name, t4.rs_type_param, t4.create_time, t4.update_time, t5.action_id, t5.action_name, t5.trigger_status, t5.trigger_action, t5.create_time, t5.update_time, t5.policy_id, t5.nf_address_list_id.").DataType("string").Required(false)).
+		Param(ws.QueryParameter("reverse", "Sort order, true-desc, false-asc.").DataType("bool").DefaultValue("false").Required(false)).
+		Param(ws.QueryParameter("offset", "Beginning index of result to return. Use this option together with limit.").DataType("uint32").Required(false)).
```
Comment (huojiao2006) [inline pkg/apis/alerting/v1alpha2/register.go]: For API doc, should we use the same explaination for the same parameters as in notification?
In nf is ://beginning data offset of this page, default 0

File: pkg/apis/alerting/v1alpha2/register.go [hunk 2]
```diff
@@ -0,0 +1,1380 @@
...
+		Writes(pb.DescribeAlertStatusResponse{}).
+		Returns(http.StatusOK, RespOK, pb.DescribeAlertStatusResponse{})).
+		Consumes(restful.MIME_JSON, constants.MIME_MERGEPATCH).
+		Produces(restful.MIME_JSON)
+
+	ws.Route(ws.GET("/nodes/alert_status").To(alerting.DescribeAlertStatusNode).
+		Doc("Describe Alert Status Node Level").
+		Param(ws.QueryParameter("alert_ids", "Specify alert ids to query, comma-separated, eg. al-WgBBMmRv3rMP,al-QVjOxnkD3mwW.").DataType("string").Required(false)).
+		Param(ws.QueryParameter("alert_names", "Specify alert names to query, comma-separated, eg. alert-1,alert-2.").DataType("string").Required(false)).
+		Param(ws.QueryParameter("disables", "Specify alert disab
```
Comment (huojiao2006) [inline pkg/apis/alerting/v1alpha2/register.go]: Chinese?

File: pkg/apiserver/alerting/alerting.go [hunk 3]
```diff
@@ -0,0 +1,2515 @@
...
+	}
+
+	client, err := alclient.NewClient()
+	if err != nil {
+		glog.Errorf("Failed to create alert grpc client %+v.", err)
+		response.WriteAsJson(&pb.CreateResourceFilterResponse{})
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```
Comment (huojiao2006) [inline pkg/apiserver/alerting/alerting.go]: 60 could be a constant, will be better

File: pkg/apiserver/alerting/alerting.go [hunk 4]
```diff
@@ -0,0 +1,2515 @@
...
+		response.WriteAsJson(&pb.ModifyResourceFilterResponse{})
+		return
+	}
+
+	glog.Errorf("ModifyResourceFilter success: %+v", resp)
+
+	response.WriteAsJson(resp)
+}
+
+func DeleteResourceFilters(request *restful.Request, response *restful.Response) {
```
Comment (huojiao2006) [inline pkg/apiserver/alerting/alerting.go]: no error need to return?

File: pkg/models/alert/query_setting.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 The KubeSphere Authors. All rights reserved.
+// Use of this source code is governed by a Apache license
+// that can be found in the LICENSE file.
+
+package models
+
+var SearchWordColumnTable = []string{
```
Comment (huojiao2006) [inline pkg/models/alert/query_setting.go]: This file is necessary?

File: pkg/apiserver/alerting/alerting.go (hunk 3 above)
Comment (Ma-Dan) [reply pkg/apiserver/alerting/alerting.go]: Fixed

File: pkg/apiserver/alerting/alerting.go (hunk 4 above)
Comment (Ma-Dan) [reply pkg/apiserver/alerting/alerting.go]: Fixed

**Pull request closed without merging**